### PR TITLE
Fixed a runtime error caused by the Pills Container component trying to access an undefined icon.

### LINKED ISSUE #3138: The Pills Container component tries to access a property of undefined (runtime major crash)
**Error summary**: the [Pills Component](https://react.lightningdesignsystem.com/components/pill-container/) causes a major crash / runtime error

**Where the error occurs**:
design-system-react\components\pill-container\private\selected-listbox.jsx
at **line 149**, function **getIcon**

**Error description**: The function getIcon (pic 1) seems to handle the icon display for the currently selected option. Option === current selection, which legitimately can be undefined according to the component props (pic 2). The code tries to access the property option.icon and this results in an error "Cannot access property icon of undefined"

**Proposed solution**: since both selection and icon are optional props (pic 2), either change the props typings OR use optional chaining (?.) operator in the code. From the looks of it, the latter is preferable since the code handles the possible undefined everywhere except this one place. Seems like just a simple oversight but causes a big trouble in the app
image
image

### PR DESCRIPTION
**Pull request by shadwar123**
Fixes #3138

### Additional description

The Pills Container component was encountering a runtime error because it tried to access an icon that was sometimes undefined. To fix this, I applied optional chaining where the icon is being updated. If the icon is not present, we set its value to null. As a result, the getAvatar() function is called, and the props are passed with icon = null and avatar = value. In the Utilities, there's a Pill component that renders either the icon or the avatar based on their values using render() function.
@interactivellama 

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [ ] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [ ] `npm run lint:fix` has been run and linting passes.
* [ ] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] CircleCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

**Comment by interactivellama** [comment]
Thanks for picking this up! Please remove the package lock file changes unless you are updating dependencies.

File: components/pill/index.jsx [hunk 1]
```diff
@@ -198,7 +198,7 @@ class Pill extends React.Component {
 	};
 
 	renderIcon = () => {
-		const icon = this.props.icon || this.props.avatar;
+		const icon = this.props?.icon || this.props?.avatar;
```
Comment (interactivellama) [inline components/pill/index.jsx:201]: What is the use case for `this.props` not being defined?

Comment (shadwar123) [reply components/pill/index.jsx:201]: Actually the undefined problem is comming beacuse of selected-listbox.jsx getIcon funtion as In some cases option only have icon or in other case only avatar, there i have applied const iconObject = option?.icon || null; and here const icon = this.props?.icon || this.props?.avatar; for the extra safety.

**Comment by interactivellama** [comment]
I see that you downgraded the package lock verison. Please remove your package.lock file.

**Comment by shadwar123** [comment]
> I see that you downgraded the package lock verison. Please remove your package.lock file.

Done,
Actually to run the program i have to run this npm install --legacy-peer-deps which changes Package-lock.json file. 
Now I have removed the Package-lock.json file.

**Comment by interactivellama** [comment]
@shadwar123 You shouldn't delete package-lock files in libraries you are contributing to. Please restore the old version.

**Comment by shadwar123** [comment]
> @shadwar123 You shouldn't delete package-lock files in libraries you are contributing to. Please restore the old version.

Done.

**Review by interactivellama — APPROVED** [review]

**Pull request merged**